29 useractivitylog model implementation with prisma and supabase

alexchris99 opened the pull request:
# Summary
created userActivitylog to keep track of the user interaction in the web page using: node, express, prisma and supabase.

# Likined issue
Closes #29

# Changes
- Created Question model with Prisma schema
- Implemented Log controller with CRUD operations
- Added Log routes
- Set up relationship between Log, Quizz, User and Questions
# Comments
I think the way I work with the info is the correct but In case that not can you make me coments so I can corrected

kazuki-0418 committed:
Merge pull request #39 from Cornerstone-CICCC/2-user-model-implementation-with-prisma-and-supabase-using-mvc-architecture

2 user model implementation with prisma and supabase using mvc architecture

Commit:
Created UserActivitylog class

Commit:
Feature User-Activity-Log

Commit:
Feature User-Activity-Log

Commit:
Feature User-Activity-Log

kazuki-0418 commented on backend/src/index.ts:
```diff
@@ -62,6 +63,7 @@ app.options("*", cors());
 app.use("/api/users", userRouter);
 app.use("/api/quizzes", quizRouter);
 app.use("/api/questions", questionRouter);
+app.use("/log", logRouter);
```
Can you fix this endpoint from "/log" to "/api/user_activity_logs"

kazuki-0418 replied:
and also change name logRouter to activityLogRouter

kazuki-0418 commented on backend/src/models/userActivityLog.model.ts:
```diff
@@ -0,0 +1,56 @@
+          date: new Date(),
+          questionsAnswered: data.questionsAnswered,
+          correctAnswers: data.correctAnswers,
+          // sessionsJoined: TODO
```
Why did you comment out here?
It is also supposed to be updated from frontend
`// sessionsJoined: TODO`

kazuki-0418 commented on backend/src/models/userActivityLog.model.ts:
```diff
@@ -0,0 +1,56 @@
+    }
+  }
+
+  async getLogById(id: string) {
```
change name getLogById to getActivityLogsByUserId
and also change argument id to userId

kazuki-0418 commented on backend/prisma/schema.prisma:
```diff
@@ -61,7 +61,7 @@ model UserActivityLog {
     sessionsJoined   Int       @default(0)
     
-    @@unique([userId, date])
+    @@unique([userId, date, id])
```
It is not supposed to be necessary when creating activity log
So we don't need an id for uniqueness

Commit:
Fixed all the requested changues

alexchris99 commented:
# Sumary
- Fixed naming standars
- Fixed activityUserLogModel
- Argument id changued to userId
- Remove id from prisma because uniquenes its not necessary

kazuki-0418 commented on backend/src/types/userActivityLog.ts:
```diff
@@ -0,0 +1,9 @@
+export type userActivityLog = {
```
Create Types createActivityLogs and updateActivityLogs?
references as types/question.ts

kazuki-0418 commented on backend/src/controllers/userActivityLog.controller.ts:
```diff
@@ -0,0 +1,56 @@
+    }
+  }
+
+  async getLogById(req: Request<{ userId: string }>, res: Response) {
```
change name getLogById to getUserActivityLogById

kazuki-0418 commented on backend/src/controllers/userActivityLog.controller.ts:
```diff
@@ -0,0 +1,56 @@
+    }
+  }
+
+  async updateActivityLog(req: Request<{ id: string }>, res: Response) {
```
change name updateActivityLog to updateUserActivityLog

kazuki-0418 commented on backend/src/models/userActivityLog.model.ts:
```diff
@@ -0,0 +1,61 @@
+      const updateLog = await prisma.userActivityLog.update({
+        where: { id },
+        data: {
+          date: new Date(),
```
You don't have to update date, so can you remove it?

kazuki-0418 commented on backend/src/models/userActivityLog.model.ts:
```diff
@@ -0,0 +1,61 @@
+        where: {
+          userId_date: {
+            userId,
+            date: new Date(),
```
Can you remove date: new Date() line:52 
and also Can you add lastActivityDate

kazuki-0418 reviewed (changes requested):
Can you change name of field "date" to "lastActivityDate" in UserActivityLog prisma.schema?
And change type Date to string 
It is supposed to be YYYY-MM-DD format

kazuki-0418 committed:
fix conflicts

Commit:
Requested changues

Commit:
Requested Changues

Commit:
Delete folder dist

alexchris99 commented:
# Summary 
- Requested changues applied

kazuki-0418 commented on backend/src/routes/userActivityLog.routes.ts:
```diff
@@ -0,0 +1,15 @@
+
+const activityLogRouter = Router();
+
+// create user log
```
please fix comments user log to user activity log

Commit:
Fixed getUserActivityLogById

Commit:
Comments fixed

Commit:
Dist directory removed

kazuki-0418 reviewed (approved):
LGTM
